proto/fulfillment/v1/cluster_type.proto

```diff
@@ -33,9 +33,57 @@ message Cluster {
 }
 
 // The spec contains the details of a cluster as desired by the user.
-//
-// Note that currently this is empty because there are no properties of the cluster that can be modified by the user.
-message ClusterSpec {}
+message ClusterSpec {
+  // Desired node sets of the cluster.
+  //
+  // This will be automatically set by the system when the cluster is initially created, according to the template
+  // selected by the user, and can be later modified to change the size.
+  //
+  // The key of the map is the unique identifier of the node set for this cluster.
+  //
+  // For example, a cluster created with two different node sets, one for nodes without GPUs and another for nodes with
+  // GPUs could be represented like this:
+  //
+  // ```json
+  // {
+  //   "id": "123",
+  //   "spec": {
+  //     "node_sets": {
+  //       "compute": {
```

Comment:
In this case, is "compute" just an arbitrary name assigned to this particular node set?

If so, it would be much more k8s-idiomatic to make this an array of node sets, and make each one have an explicit "name" field. https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#lists-of-named-subobjects-preferred-over-maps

Reply:
Yes, "compute" is a name assigned by the system to this node set.

What you suggest is what I did in the first iteration of this patch, but as I explained [here](https://github.com/innabox/fulfillment-api/pull/25#issuecomment-2820867281) protobuf an gRPC it complicates things because the recommended mechanism for that is the [field mask](https://protobuf.dev/reference/protobuf/google.protobuf/#field-mask) and that doesn't support updating individual elements of arrays.

We can anyhow use the array if we want, it will just be a bit more complicated to implement, as the server will need to learn that this specific fields needs special handling.

Reply:
I've opened https://github.com/innabox/fulfillment-api/issues/26 related to this line of questions.